src/lib.rs

```diff
@@ -1837,24 +1837,25 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 
 		// First add all connected peers, preferring to list the connected address if available.
 		let connected_peers = self.peer_manager.get_peer_node_ids();
-		for (node_id, con_addr_opt) in connected_peers {
+		for (node_id, con_addr_opt) in &connected_peers {
 			let stored_peer = self.peer_store.get_peer(&node_id);
 			let stored_addr_opt = stored_peer.as_ref().map(|p| p.address.clone());
+
 			let address = match (con_addr_opt, stored_addr_opt) {
-				(Some(con_addr), _) => NetAddress(con_addr),
+				(Some(con_addr), _) => NetAddress(con_addr.clone()),
```

Comment:
If you store the number of connected peers, you can avoid this clone by using `peers.iter().take(len)` below and keeping the code as it originally was.